README.rst

```diff
@@ -2,3 +2,117 @@ cdist manifestation
 ===================
 
 A helper for writing cdist manifests in Python.
+
+Motivation
+----------
+
+Writing complex manifests in shell can be painful and laborious. With Python and
+a proper abstraction layer, we can write manifests more easily.
+
+Usage
+-----
+
+First of all make sure your manifest is executable and contains
+``#!/usr/bin/env python`` shebang.
+
+Types
+`````
+
+To use type ``__directory``, import ``directory`` (without leading underscores)
+from ``cdist_manifestation.types`` like this:
+
+.. code-block:: python
+
+    from cdist_manifestation.types import directory
+
+Every type is a function accepting a single positional argument, ``object_id``
+which is optional, and arbitrary amount of keyword arguments, which are the
+type parameters.
+
+So, one could create a directory like this:
+
+.. code-block:: python
+
+    from cdist_manifestation.types import directory
+
+    directory('/my/dir', mode=755, owner='me', parents=True)
+
+This will translate to ``__directory /my/dir --mode 755 --owner me --parents``.
+
+There are two noteworthy things in this example:
+
+1. Parameter values will be converted to strings implicitly
+2. We use booleans to define boolean parameters
+
+When you want to use the same parameter multiple times, just provide an iterable
+like this:
+
+.. code-block:: python
+
+    from cdist_manifestation.types import user_groups
+
+    user_groups('me', group=['group1', 'group2'])
+
+This will translate to ``__user_groups me --group group1 --group group2``.
+
```

Comment:
Here would be a good place to mention *all* rules about kwargs parameter handling:

  - kwargs of the imported function become parameters of the cdist object definition
    - Underscores in those args will be implicitly converted to dashes
    - Values of these arguments are implicitly converted to strings, except:
      - Booleans, which simply denote a presence of a boolean parameter
      - Iterables, which denote presence of multiple instances of the same parameter. Values returned by this iterable will be implicitly converted to strings disregarding these exceptions.

Reply:
Done.